src/signalo/core/models.py

```diff
@@ -71,3 +96,31 @@ class Sign(ComputedFieldsModel):
     )
     def geom(self):
         return self.geom
+
+
+@receiver(signals.pre_delete, sender=Sign)
+@transaction.atomic()
+def ensure_sign_order_on_delete(sender, instance, *args, **kwargs):
+    """
+    Ensure dense ranking of Sign orders in spite of deletion of related Azimuth
+    """
+    azimuths = Azimuth.objects.all()
```

Comment:
the azimuth exists on the sign instance, so there should be no need to look for azimuths and/or poles.

You fetch all the signs having the same azimuth instance except the one being deleted and you reset the order. Does this make sense?

Reply:
Sense made = 0, my bad I was probably distracted by the ambiant sound. Hope it's better now.